vmdk_plugin/main_windows.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// A VMDK Docker Data Volume plugin implementation for windows.
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/go-winio"
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/go-plugins-helpers/volume"
+)
+
+const (
+	npipeAddr = `\\.\pipe\vsphere-dvs` // Plugin's npipe address
+)
+
+var (
+	mountRoot = filepath.Join(os.Getenv("LOCALAPPDATA"), "docker-volume-vsphere", "mounts") // VMDK volumes are mounted here
+)
+
+// NpipePluginServer serves HTTP requests from Docker over windows npipe.
+type NpipePluginServer struct {
+	PluginServer
+	driver   *volume.Driver // The driver implementation
+	mux      *http.ServeMux // The HTTP mux
+	listener net.Listener   // The npipe listener
+}
+
+// Response for /Plugin.Activate to activate the Docker plugin.
+type PluginActivateResponse struct {
+	Implements []string // Slice of implemented driver types
```

Comment:
This is defined as per the [Docker API](https://docs.docker.com/engine/extend/plugin_api/#pluginactivate), so we might not be able to change it.